### PR TITLE
Add decoding=async attribute to images by default

### PR DESCRIPTION
**Pull request by westonruter**
Note that most of the test changes are to make testing work on more environments, specifically where the wp-content directory is not in the standard location.

Trac ticket: https://core.trac.wordpress.org/ticket/53232

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/media.php [hunk 1]
```diff
@@ -1076,6 +1076,8 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 			}
 		}
 
+		$attr['decoding'] = 'async';
```
Comment (westonruter) [inline src/wp-includes/media.php:1079]: I suppose the question will be whether this should have a filter like lazy-loading does via `wp_lazy_loading_enabled()`. Should there be a `wp_async_decoding_enabled()` function and filter passed the same args?

Comment (westonruter) [reply src/wp-includes/media.php:1079]: In contrast with lazy-loading in which `lazy` is omitted for images likely to be in the first viewport, the `decoding=async` attribute would still be added to all images.

**Comment by mehigh** [comment]
Have done the tests and covered the potential caveats I could think about, and this is passing with flying colors, as the lab+local tests have contradicted some of the conclusions I've drawn after reading the [specifications](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-decoding).

Full result of the research posted on Trac :
https://core.trac.wordpress.org/ticket/53232#comment:34

TL;DR:
Explored several edge-cases in order to find potential downsides from including this attribute. Used webpagetest, GPSI & local lighthouse tests for all tests to confirm conclusions. Used performance profiling to go deeper and confirm the data/conclusions when the results were too consistent in the lab tests.
A big thumbs-up for this to be the default option for all images, not just the ones in the_content.

**Comment by westonruter** [comment]
Closing in favor of https://github.com/WordPress/wordpress-develop/pull/2409

**Pull request closed without merging**